Nasal/mooring.nas

```diff
@@ -20,9 +20,25 @@
 #init if allowed and called for
 setlistener("/sim/signals/fdm-initialized",
     func {
-        if (getprop("/controls/mooring/automatic") and getprop("/controls/mooring/allowed")) {
-            seaplane = Mooring.new();
+
+        var presets = props.globals.getNode("/sim/presets");
+        var seaplanes = props.globals.getNode("/systems/mooring/route").getChildren("seaplane");
+        var harbour = "";
+        var airport = presets.getChild("airport-id").getValue();
+        setprop("/controls/mooring/port-available", 0);
+        if(airport != nil and airport != "") {
+            for(var i=0; i<size(seaplanes); i=i+1) {
+                harbour = seaplanes[ i ].getChild("airport-id").getValue();
+                if(harbour == airport) {
+                    setprop("/controls/mooring/port-available", 1);
+                }
+            }
         }
+        settimer(func{
```

Comment:
What's the purpose of waiting a second here?

Reply:
The new logic is used to set the "/controls/mooring/port-available" property. That property is used as a condition for the availability of the "Go now" button and also used in Systems/bushkit.xml mooring-allowed switch.
I was unable to leverage the second block of logic, which basically does the same thing, as it was only called when either the user ("Go now" button) or the system (on-reload) was actually invoked. I needed it available at all times for the two previously mentioned uses.